apollo-normalized-cache/src/commonMain/kotlin/com/apollographql/apollo/cache/normalized/internal/SortedInputFieldMapWriter.kt

```diff
@@ -3,18 +3,14 @@ package com.apollographql.apollo.cache.normalized.internal
 import com.apollographql.apollo.api.ScalarType
 import com.apollographql.apollo.api.internal.InputFieldMarshaller
 import com.apollographql.apollo.api.internal.InputFieldWriter
-import com.apollographql.apollo.api.internal.Utils.__checkNotNull
-import java.io.IOException
-import java.util.ArrayList
-import java.util.Collections
-import java.util.Comparator
-import java.util.TreeMap
+import com.apollographql.apollo.api.internal.Throws
+import okio.IOException
 
-class SortedInputFieldMapWriter(private val fieldNameComparator: Comparator<String>) : InputFieldWriter {
-  private val buffer = TreeMap<String, Any?>(fieldNameComparator)
+class SortedInputFieldMapWriter() : InputFieldWriter {
+  private val buffer = mutableMapOf<String, Any?>()
 
   fun map(): Map<String, Any?> {
-    return Collections.unmodifiableMap(buffer)
+    return buffer.toList().sortedBy { it.first }.toMap()
```

Comment:
nit: should `buffer.toSortedMap()` do the same?

Reply:
I didn't find a `toSortedMap` for iOS :(